crates/solidity/inputs/language/src/grammar.rs

```diff
@@ -322,8 +322,7 @@ fn resolve_grammar_element(ident: &Identifier, ctx: &mut ResolveCtx) -> GrammarE
             let thunk = Rc::new(NamedParserThunk {
                 name: ident.to_string().leak(),
                 context: lex_ctx,
-                // Enums have a single reference per variant, so they should be inlined.
-                is_inline: matches!(elem.as_ref(), Item::Enum { .. }),
+                is_inline: false,
```

Comment:
I would like to get @AntonyBlakey's eyes on this one. Some of these nodes look very useful, for example `ContractMember` and `ConstructorAttribute`, as it makes it easier to find/match on these elements.

However, some of them look extraneous indeed, and are only there for purpose of authoring the grammar/versioning. For example:

- `ElementaryType` and `YulLiteral` which will (almost) always have a unique parent that can be matched against. Maybe we can refactor the grammar a bit to make this more accurate?

- `TypedTupleMember` and `UntypedTupleMember` which only exist to make parsing/backtracking correct, but provide no additional meaning. Not sure how to avoid it.

What do you think?

I’m trying to avoid adding optional inlining to the DSL unless we absolutely need it. As based on all of our new design decisions, and AST structure, it will make it less obvious to go from grammar to CST/AST, and add another layer of complexity that users have to deal with ..

So, if we are happy with these changes, I’m fine with merging the PR as-is for now, and I can manually go over the enums to see if any of them can be better structured. For example, inlining something like `ElementaryType` variants into the types it references, since it is almost always used inside another Enum. It will probably be more nuanced than that though.

Reply:
Could we make the non-interesting enum into something else in the grammar i.e. effectively add a non-outlined characteristic by using a different name for the concept. It sure sounds like it has a different purpose.

Reply:
f2f: we think the PR is good enough for now, and let's review the added kinds later to see if anything can be pruned/removed.